### PR TITLE
Add new e2e tests for the checks catalog view

### PR DESCRIPTION
**Pull request by arbulu89**
New cypress tests feature for the Checks catalog.

It checks:
- The checks catalog is properly populated
- The displayed groups and checks are the correct ones
- The check detailed information box is properly extended and collapsed

### CODE REVIEW

File: test/e2e/cypress/integration/checks_catalog.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            cy.get('tr.check-row').should('have.length', 35)
+        })
+    })
+
+    describe('Checks grouping and identification is correct', () => {
+        availableChecks.forEach((checks, group) => {
+            it(`should include group '${group}'`, () => {
+                cy.get('.check-group > h4').should('contain', group)
+            })
+            checks.forEach((check_id) => {
```
Comment (nelsonkopliku) [inline test/e2e/cypress/integration/checks_catalog.js:26]: just nitpicking. camelCase `checkId` :stuck_out_tongue_winking_eye:

File: test/e2e/cypress/support/commands.js [hunk 2]
```diff
@@ -64,4 +64,24 @@ Cypress.Commands.add('loadScenario', (scenario) => {
...
+        fixturesPath,
+        webApiHost,
+        webApiPort
+    ] = [
+        Cypress.env('fixtures_path'),
+        Cypress.env('web_api_host'),
+        Cypress.env('web_api_port')
+    ]
+    cy.log(`Loading checks catalog "${catalog}"...`)
+    cy.fixture(`${catalog}`).then((file) => {
```
Comment (nelsonkopliku) [inline test/e2e/cypress/support/commands.js:80]: No need to interpolate, right? `cy.fixture(catalog)` should be enough :thinking:

File: test/e2e/cypress/support/commands.js [hunk 3]
```diff
@@ -64,4 +64,24 @@ Cypress.Commands.add('loadScenario', (scenario) => {
     ]
     cy.log(`Loading scenario "${scenario}"...`)
     cy.exec(`cd ${fixturesPath} && ${photofinishBinary} run --url "http://${collectorHost}:${collectorPort}/api/collect" ${scenario}`)
-})
\ No newline at end of file
+})
+
+Cypress.Commands.add('loadChecksCatalog', (catalog) => {
+    const [
+        fixturesPath,
```
Comment (nelsonkopliku) [inline test/e2e/cypress/support/commands.js:71]: This `fixturesPath` here maybe can go away as we're not using it.

File: test/e2e/cypress/fixtures/checks-catalog/checks_catalog.feature [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    Then the displayed checks belong properly to their own group
+
+Scenario: Check detailed information is expanded if the information icon is clicked
+    When I click the information details for each of the checks
+    Then the detailed information box is expanded
+    When I click the information details for each of the checks again
+    Then the detailed information box is collapsed
+
+Scenario: Check detailed information is expanded if catalog url included the check id
+    When I naviaate the Checks catalog adding a specific check id (/catalog#00081D)
```
Comment (nelsonkopliku) [inline test/e2e/cypress/fixtures/checks-catalog/checks_catalog.feature:23]: Typo: `navigate`

File: test/e2e/cypress/integration/checks_catalog.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                    cy.get('.check-row').should('contain', check_id)
+                })
+            })
+        })
+    })
+
+    describe('Individual checks data is expanded', () => {
+        it('should expand check data when clicked', () => {
+          const firstCheck = availableChecks.get('Corosync')[0]
+          cy.get('.check-row').filter(
```
Comment (nelsonkopliku) [inline test/e2e/cypress/integration/checks_catalog.js:37]: I am wondering if we need to get the `check-row` element and then filter by attrubute id.

We could just 
```
cy.get(`#${firstCheck}`).find('td > a.link-dark')
```

Or maybe change the html to add add a specific class or id to the `a` link element and simplify selectors.

What do you think?

**Review by nelsonkopliku — CHANGES_REQUESTED** [review]
Great job!

Just a couple of nitpicking and a typo :wink:

File: test/e2e/cypress/integration/checks_catalog.js (hunk 5 above)
Comment (arbulu89) [reply test/e2e/cypress/integration/checks_catalog.js]: Yes, more or less it works. As removing some `filters` looks like reduces the time of these operations, the test started failing, so I have added other assertion (`have.class`) to wait until this is done.
This makes the test pass.
(change done in : https://github.com/trento-project/trento/pull/736/commits/4d3b70e79a16cbb37bfb1f235f937572f4ee1452)

Comment (nelsonkopliku) [reply test/e2e/cypress/integration/checks_catalog.js]: Great!

**Review by nelsonkopliku — APPROVED** [review]

**Pull request merged**